plugins/cost-insights/src/components/AlertInsights/AlertInsights.tsx

```diff
@@ -14,31 +14,203 @@
  * limitations under the License.
  */
 
-import React from 'react';
-import { Grid } from '@material-ui/core';
-import { AlertInsightsSection } from './AlertInsightsSection';
+import React, { useCallback, useEffect, useState } from 'react';
+import pluralize from 'pluralize';
+import { Box, Grid, Snackbar } from '@material-ui/core';
+import { default as MuiAlert } from '@material-ui/lab/Alert';
+import { AlertDialog } from './AlertDialog';
+import { AlertStatusSummary } from './AlertStatusSummary';
+import { AlertStatusSummaryButton } from './AlertStatusSummaryButton';
 import { AlertInsightsHeader } from './AlertInsightsHeader';
-import { Alert } from '../../types';
+import { AlertInsightsSection } from './AlertInsightsSection';
+import {
+  useAlerts,
+  useScroll,
+  useLoading,
+  ScrollType,
+  MapLoadingToProps,
+} from '../../hooks';
+import { DefaultLoadingAction } from '../../utils/loading';
+import { Alert, AlertOptions, Maybe } from '../../types';
+import { sumOfAllAlerts } from '../../utils/alerts';
+
+type MapLoadingtoAlerts = (isLoading: boolean) => void;
 
-const title = "Your team's action items";
-const subtitle =
-  'This section outlines suggested action items your team can address to improve cloud costs.';
+const mapLoadingToAlerts: MapLoadingToProps<MapLoadingtoAlerts> = ({
+  dispatch,
+}) => (isLoading: boolean) =>
+  dispatch({ [DefaultLoadingAction.CostInsightsAlerts]: isLoading });
 
 type AlertInsightsProps = {
-  alerts: Array<Alert>;
+  group: string;
+  active: Alert[];
+  snoozed: Alert[];
```

Comment:
Do we need to pass these in ? We're already getting `alerts` from `useAlerts()` so it feels a bit weird to pass these in

Reply:
These are passed in to preserve referential equality of these snoozed, etc. alerts across components. Filtering be done inside the component but then we'd get different references. Not terrible, since they derive from `alerts.alerts` which should remain the same, but not really preferable either. 

Ideally, I'd like each component to update only when their props meaningfully change - not just whenever the parent rerenders.

 I think the root of the evil here seems to be the design of the alert state, which can be massaged a bit. I'll give it another pass and see if it allays some of these concerns.